File: mirea/cources/2026-2027/aii-july/02/Part-01-Linear-structures/examples/example-03.cpp
#include <deque>
#include <iostream>

using namespace std;

int main() {
    deque<int> d = {20, 30};

    // Deque позволяет жить на обоих концах структуры.
    d.push_front(10);
    d.push_back(40);

    cout << "front = " << d.front() << '\n';
    cout << "back = " << d.back() << '\n';

    // В C++ deque индексируется эффективно.
    cout << "middle = " << d[1] << '\n';

    d.pop_front();
    d.pop_back();

    cout << "remaining: ";
    for (int value : d) {
        cout << value << ' ';
    }
    cout << '\n';

    return 0;
}

File: mirea/cources/2026-2027/aii-july/02/Part-01-Linear-structures/examples/example-03.py
from collections import deque


d = deque([20, 30])

# Deque позволяет работать с обоими концами.
d.appendleft(10)
d.append(40)

print("front =", d[0])
print("back =", d[-1])

# Индексация есть, но для середины ею не стоит злоупотреблять.
print("middle =", d[1])

d.popleft()
d.pop()
print("remaining:", *d)